### PR TITLE
[IMP] project: add some extra fields in project populate data

### PR DESCRIPTION
**Pull request by ppr-odoo**
In this commit, we added extra fields: company, partner, users

task-3082068

### CODE REVIEW

File: addons/project/populate/project.py [hunk 1]
```diff
@@ -57,15 +57,37 @@ def get_stage_ids(random, **kwargs):
...
+        def get_partner_id(random, **kwargs):
+            if kwargs['values'].get('project_id'):
+                return random.choice(partner_per_company[kwargs['values']['company_id']])
+            else:
+                return self.env.user.partner_id.id
+        def get_company_id(random, **kwargs):
+            if kwargs['values'].get('project_id'):
+                return self.env['project.project'].browse(kwargs['values'].get('project_id')).company_id.id
+            else:
+                return self.env.company.id
```
Comment (auon-odoo) [inline addons/project/populate/project.py:87]: Actually, I wonder if company_id is needed :thinking:

File: addons/project/populate/project.py [hunk 2]
```diff
@@ -57,15 +57,37 @@ def get_stage_ids(random, **kwargs):
...
     _inherit = "project.task"
     _populate_sizes = {"small": 500, "medium": 5000, "large": 50000}
-    _populate_dependencies = ["project.project"]
+    _populate_dependencies = ["project.project", "res.partner", "res.users"]
 
     def _populate_factories(self):
         project_ids = self.env.registry.populated_models["project.project"]
         stage_ids = self.env.registry.populated_models["project.task.type"]
+        user_ids = self.env.registry.populated_models['res.users']
+        partner_per_company = {
```
Comment (auon-odoo) [inline addons/project/populate/project.py:66]:
```suggestion
        partner_ids_per_company_id = {
```
to be precise :angel:

File: addons/project/populate/project.py [hunk 3]
```diff
@@ -57,15 +57,37 @@ def get_stage_ids(random, **kwargs):
...
+        }
         def get_project_id(random, **kwargs):
             return random.choice([False, False, False] + project_ids)
         def get_stage_id(random, **kwargs):
             return random.choice([False, False] + stage_ids)
+        def get_partner_id(random, **kwargs):
+            if kwargs['values'].get('project_id'):
+                return random.choice(partner_per_company[kwargs['values']['company_id']])
+            else:
+                return self.env.user.partner_id.id
```
Comment (auon-odoo) [inline addons/project/populate/project.py:82]:
```suggestion
        def get_partner_id(random, **kwargs):
            if kwargs['values'].get('project_id'):
                partners = partner_per_company[kwargs['values']['company_id']]
                return random.choice(partners + [False] * len(partners))
            else:
                return False
```
- If the task is private, we can just set the partner to False.
- Let's also have tasks with no partner. :+1:

**Review by auon-odoo — CHANGES_REQUESTED** [review]
Thank you! :fire:

File: addons/project/populate/project.py (hunk 3 above)
Comment (auon-odoo) [reply addons/project/populate/project.py:82]: and so `partner_ids`

**Review by auon-odoo — APPROVED** [review]
Thanks a lot for your work! :fire:

File: addons/project/populate/project.py [hunk 4]
```diff
@@ -57,15 +57,43 @@ def get_stage_ids(random, **kwargs):
...
     def _populate_factories(self):
         project_ids = self.env.registry.populated_models["project.project"]
         stage_ids = self.env.registry.populated_models["project.task.type"]
+        user_ids = self.env.registry.populated_models['res.users']
+        partner_ids_per_company_id = {
+            group['company_id'][0]: group['ids']
+            for group in self.env['res.partner']._read_group(
+                domain=[('company_id', '!=', False), ('id', 'in', self.env.registry.populated_models["res.partner"])],
+                fields=['company_id', 'ids:array_agg(id)'],
+                groupby=['company_id'],
```
Comment (xavierbol) [inline addons/project/populate/project.py:71]: No need to name the parameters if the parameters are given in the right orders. 🙂 
```suggestion
                [('company_id', '!=', False), ('id', 'in', self.env.registry.populated_models["res.partner"])],
                ['company_id', 'ids:array_agg(id)'],
                ['company_id'],
```

File: addons/project/populate/project.py [hunk 5]
```diff
@@ -57,15 +57,43 @@ def get_stage_ids(random, **kwargs):
...
+            )
+        }
+        company_id_per_project_id = {
+            record['id']: record['company_id']
+            for record in self.env['project.project'].search_read(
+                domain=[('company_id', '!=', False), ('id', 'in', self.env.registry.populated_models["project.project"])],
+                fields=['company_id'],
```
Comment (xavierbol) [inline addons/project/populate/project.py:78]: Same comment here. 🙂 

```suggestion
                [('company_id', '!=', False), ('id', 'in', self.env.registry.populated_models["project.project"])],
                ['company_id'],
```

**Review by xavierbol — COMMENTED** [review]
Thanks for your work, I left 2 comments and after that it will look good to me. 🙂 

Can you also rebase your branch to be up to date (except if it is already done)? Thanks in advance! 🙂

**Comment by ppr-odoo** [comment]
@xavierbol Done the changes.thanks

**Review by xavierbol — APPROVED** [review]
LGTM, thanks for your work. 🙂 

robodoo r+

**Pull request closed without merging**